pkg/controllermanager/controller/seed/seed_reconcile.go

```diff
@@ -164,18 +162,11 @@ func (r *reconciler) syncGardenSecrets(ctx context.Context, gardenClient kuberne
 				},
 				Data: secret.Data,
 			}
-			// TODO: use controllerutil.CreateOrUpdate once cache is graduated to beta https://github.com/gardener/gardener/issues/2822
-			if err := gardenClient.Client().Create(ctx, seedSecret); err != nil {
-				if !apierrors.IsAlreadyExists(err) {
-					return err
-				}
-				_, err2 := kutil.TryUpdateSecret(ctx, gardenClient.Kubernetes(), retry.DefaultRetry, seedSecret.ObjectMeta, func(obj *corev1.Secret) (*corev1.Secret, error) {
-					obj.Annotations = seedSecret.Annotations
-					obj.Labels = seedSecret.Labels
-					obj.Data = seedSecret.Data
-					return obj, nil
-				})
-				return err2
+
+			if _, err := controllerutil.CreateOrUpdate(ctx, gardenClient.Client(), seedSecret, func() error {
+				return nil
```

Comment:
```suggestion
				seedSecret.Annotations = secret.Annotations
				seedSecret.Labels = secret.Labels
				seedSecret.Data = secret.Data
				return nil
```

and remove those from above `seedSecret`

Reply:
Done.